app/javascript/chat/util.js

```diff
@@ -76,39 +81,41 @@ export function adjustTimestamp(timestamp) {
   return time;
 }
 
-
 export function setupNotifications() {
-  navigator.serviceWorker.ready.then((serviceWorkerRegistration) => {
-    serviceWorkerRegistration.pushManager.getSubscription()
-      .then(function(subscription) {
+  navigator.serviceWorker.ready.then(serviceWorkerRegistration => {
+    serviceWorkerRegistration.pushManager
+      .getSubscription()
+      .then(subscription => {
         if (subscription) {
           return subscription;
         }
         return serviceWorkerRegistration.pushManager.subscribe({
           userVisibleOnly: true,
-          applicationServerKey: window.vapidPublicKey
+          applicationServerKey: window.vapidPublicKey,
         });
-      }).then(function(subscription) {
-        sendKeys(subscription.toJSON(), null, null)
+      })
+      .then(subscription => {
+        sendKeys(subscription.toJSON(), null, null);
       });
   });
 }
 
 export function getNotificationState() {
-  //Not yet ready
-  if (!window.location.href.includes('ask-for-notifications')){
-    return "dont-ask"
+  // Not yet ready
+  if (!window.location.href.includes('ask-for-notifications')) {
+    return 'dont-ask';
   }
 
   // Let's check if the browser supports notifications
-  if (!("Notification" in window)) {
-    return "not-supported"
-  }  else if (Notification.permission === "granted") {
+  if (!('Notification' in window)) {
+    return 'not-supported';
+  }
+
+  const { permission } = Notification;
+
+  if (permission === 'granted') {
     setupNotifications();
-    return "granted"
-  }  else if (Notification.permission !== 'denied') {
-    return "waiting-permission"
-  } else {
-    return "denied"
   }
-}
\ No newline at end of file
+
+  return permission === 'default' ? 'waiting-permission' : permission;
```

Comment:
Since there are only three possible permission types, https://developer.mozilla.org/en-US/docs/Web/API/Notification/permission, I'm assuming when it's `'default'` @benhalpern, you wish it to be set to `'waiting-permission'`?

Reply:
Yeah.